File: examples/multiple-files.js
// Пример: отправка нескольких файлов и текстового сообщения
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

async function main() {
    const apiKey = 'YOUR_API_KEY'; // замените на ваш ключ

    // Формируем multipart/form-data запрос
    const formData = new FormData();
    // Добавляем файлы
    const file1 = fs.readFileSync(path.join(__dirname, 'basic.js'));
    const file2 = fs.readFileSync(path.join(__dirname, 'basic.py'));
    formData.append('files', new Blob([file1]), 'basic.js');
    formData.append('files', new Blob([file2]), 'basic.py');
    // Добавляем JSON данные
    const payload = {
        messages: [{ role: 'user', content: 'Сравните эти два файла' }],
        extra_body: { deepthink: false, web_search: false }
    };
    formData.append('data', JSON.stringify(payload));

    const response = await fetch('http://localhost:3000/v1/chat/completions', {
        method: 'POST',
        headers: {
            'Authorization': `Bearer ${apiKey}`,
        },
        body: formData
    });

    const data = await response.json();
    console.log(data.choices[0].message.content);
}

main().catch(console.error);

File: examples/multiple-files.py
import requests
import json

api_key = "YOUR_API_KEY"
url = "http://localhost:3000/v1/chat/completions"

files = [
    ('files', ('basic.js', open('examples/basic.js', 'rb'), 'application/javascript')),
    ('files', ('basic.py', open('examples/basic.py', 'rb'), 'text/x-python'))
]
data = {
    'data': json.dumps({
        "messages": [{"role": "user", "content": "Сравните эти два файла"}],
        "extra_body": {"deepthink": False, "web_search": False}
    })
}

response = requests.post(url, headers={"Authorization": f"Bearer {api_key}"}, files=files, data=data)
print(response.json()['choices'][0]['message']['content'])